src/pollManager.ts

```diff
@@ -0,0 +1,50 @@
+import { Logger } from '@map-colonies/js-logger';
+import { inject, singleton } from 'tsyringe';
+import { Watcher } from './watcher';
+import { Readiness } from './probe/readiness';
+import { Liveness } from './probe/liveness';
+import { Services } from './common/constants';
+import { IPollConfig } from './common/interfaces';
+
+@singleton()
+export class PollManager {
+  public constructor(
+    @inject(Services.LOGGER) private readonly logger: Logger,
+    @inject(Services.POLLCONFIG) private readonly pollCofig: IPollConfig,
+    private readonly watcher: Watcher,
+    private readonly liveness: Liveness,
+    private readonly readiness: Readiness
+  ) {}
+
+  public async poll(): Promise<void> {
+    const frequencyTimeOutMS = this.pollCofig.timeout.frequencyMilliseconds;
+    const livenessKillTimeOutMS = this.pollCofig.timeout.livenessKillSeconds;
+    try {
+      if (!(await this.watcher.isUpdated())) {
+        await this.delay(this.getRandomInteger());
+        this.readiness.kill();
+        await this.delay(livenessKillTimeOutMS);
+        this.liveness.kill();
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        const stackMessage = error.stack !== undefined ? error.stack : '';
+        this.logger.error(`Error occurred during poll check: ${error.message}, stack: ${stackMessage}`);
+      }
+    }
+    setTimeout(() => {
+      void this.poll();
+    }, frequencyTimeOutMS);
+  }
+
+  public async delay(ms: number): Promise<void> {
+    const msToSeconds = 1000;
```

Comment:
it's seconds to ms

Reply:
its same thing - multiply action (ms * msToSeconds) means its msToSeconds , if it was division act then its could be seconds to ms